pkg/apis/ceph.rook.io/v1/topic.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// compile-time assertions ensures CephBucketTopic implements webhook.Validator so a webhook builder
+// will be registered for the validating webhook.
+var _ webhook.Validator = &CephBucketTopic{}
+
+func ValidateHTTPSpec(s *HTTPEndpointSpec) error {
```

Comment:
I am just wondering if it is TODO?

Reply:
no. all validations are done in the generated code - will add a comment